### DISCUSSION TITLE
GDScript completion does not insert anything in some scenarios

### DISCUSSION THREAD

**Discussion by rcorre** [helix-editor/helix#10371] [Troubleshooting]
### Summary

Given the following script:

```
extends Spatial

func _ready():
	var data := var2bytes(self).compress(File.COM
```

Completions selected after `COM` will not insert anything.

### Reproduction Steps

 

I tried this:

1. `hx Example.gd`
2. Move the cursor after "COM"
3. Press `Tab` a few times.

I expected this to happen:
Selected completions are inserted.

Instead, this happened:
Completions are selected but nothing is inserted.

https://asciinema.org/a/kcE7KPiAl1mogNYc4SQxDgBZx

(neovim shown at the end for comparison)

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>
[helix.log](https://github.com/helix-editor/helix/files/10774292/helix.log)

### Platform

Linux

### Terminal Emulator

st 0.8.5

### Helix Version

22.12-256-g3b301a9d

**Comment by rcorre** [comment]
I'm wondering if this could be a bug in Godot's LSP. In the case above, the response looks like:

```
{"data":"File::COMPRESSION_GZIP","insertText":"","kind":21,"label":"COMPRESSION_GZIP"}
```

However, if I type: `var c := File.COM<TAB>`, the response looks like:

```
{"data":{"position":{"character":15,"line":108},"textDocument":{"uri":"file:///home/rcorre/src/agos/singleton/SteamAPI.gd"}},"insertText":"COMPRESSION_ZSTD","kind":21,"label":"COMPRESSION_ZSTD"}```

**Comment by erasin** [comment]
I use GODOT 4.0.3.stable.official ,
helix (master > 23.05)

The LSP runs properly. 


![WX20230602-114822](https://github.com/helix-editor/helix/assets/716514/12c14b03-a941-42ac-8027-798f656a96b2)

**Comment by rcorre** [comment]
On `godot 4.0.3.stable.arch_linux` `helix 23.03`, I still see this behavior.

@erasin when you type `TAB`, does it actually insert `COMPRESSION_FASTLZ`? For me it just clears `com`.

[helix.log](https://github.com/helix-editor/helix/files/11644352/helix.log)
